code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm

```diff
@@ -26,25 +44,44 @@ insert ascii eagle on american flag background here
 		/obj/item/device/multitool,
 		/obj/item/weldingtool,
 		/obj/item/reagent_containers/glass,
+		/obj/item/reagent_containers/syringe,
+		/obj/item/reagent_containers/food/condiment,
 		/obj/item/storage/part_replacer))
+	var/datum/looping_sound/deep_fryer/fry_loop
 
 /obj/machinery/deepfryer/Initialize()
 	. = ..()
 	create_reagents(50)
-	reagents.add_reagent("nutriment", 25)
+	reagents.add_reagent("cooking_oil", 25)
 	component_parts = list()
 	component_parts += new /obj/item/circuitboard/machine/deep_fryer(null)
 	component_parts += new /obj/item/stock_parts/micro_laser(null)
 	RefreshParts()
+	fry_loop = new(list(src), FALSE)
+
+/obj/machinery/deepfryer/RefreshParts()
+	var/oil_efficiency
+	for(var/obj/item/stock_parts/micro_laser/M in component_parts)
+		oil_efficiency += M.rating
+	oil_use = initial(oil_use) - (oil_efficiency * 0.0095)
+	fry_speed = oil_efficiency
 
 /obj/machinery/deepfryer/examine()
 	..()
 	if(frying)
 		to_chat(usr, "You can make out \a [frying] in the oil.")
 
 /obj/machinery/deepfryer/attackby(obj/item/I, mob/user)
-	if(!reagents.total_volume)
-		to_chat(user, "There's nothing to fry with in [src]!")
+	if(istype(I, /obj/item/reagent_containers/pill))
+		if(!reagents.total_volume)
+			to_chat(user, "<span class='warning'>There's nothing to dissolve [I] in!</span>")
+			return
+		user.visible_message("<span class='notice'>[user] drops [I] into [src].</span>", "<span class='notice'>You dissolve [I] in [src].</span>")
+		I.reagents.trans_to(src, I.reagents.total_volume)
+		qdel(I)
+		return
+	if(!reagents.has_reagent("cooking_oil"))
```

Comment:
:-1:

you should be able to fry with any reagent

Reply:
I don't think that's how frying works.

Reply:
realism is dumb and you can't eat deep fried metal rods IRL either

Reply:
That's was you who added it though...

Reply:
i know, i'm saying it's dumb to disallow it because reels